tests/UploadReadbackTest.cpp

```diff
@@ -382,4 +382,101 @@ INSTANTIATE_TEST_SUITE_P(PerCommandQueueType,
                                          CommandQueueType::Compute,
                                          CommandQueueType::Copy));
 
+struct ScalarBlockLayoutTests : public VexTestParam<ShaderCompilerBackend>
+{
+    struct WeirdlyPackedData
+    {
+        float vector1[3];
+        float vector2[3];
+        float vector3[3];
+    };
+};
+
+TEST_P(ScalarBlockLayoutTests, ComputeMisaligneData)
+{
+    auto ctx = graphics.BeginScopedCommandContext(CommandQueueType::Compute, SubmissionPolicy::Immediate);
+
+    std::array<WeirdlyPackedData, 13> data;
+    for (int i = 0; i < 13; ++i)
+    {
+        int v1 = 1, v2 = 4, v3 = 7;
+        for (int j = 0; j < 3; ++j)
+        {
+            data[i].vector1[j] = v1++;
+            data[i].vector2[j] = v2++;
+            data[i].vector3[j] = v3++;
+        }
+    }
+
+    Buffer dataBuffer = graphics.CreateBuffer(
+        BufferDescription{ .name = "DataBuffer", .byteSize = data.size() * sizeof(WeirdlyPackedData) });
+    Buffer resultBuffer =
+        graphics.CreateBuffer(BufferDescription{ .name = "ResultBuffer",
+                                                 .byteSize = 3 * sizeof(float),
+                                                 .usage = BufferUsage::GenericBuffer | BufferUsage::ReadWriteBuffer });
+
+    ctx.EnqueueDataUpload(dataBuffer, std::as_bytes(std::span{ data }));
+
+    ResourceBinding bindings[]{
+        BufferBinding{ .buffer = dataBuffer,
+                       .usage = BufferBindingUsage::StructuredBuffer,
+                       .strideByteSize = sizeof(WeirdlyPackedData) },
+        BufferBinding{
+            .buffer = resultBuffer,
+            .usage = BufferBindingUsage::RWStructuredBuffer,
+            .strideByteSize = 3 * sizeof(float),
+        },
+    };
+    std::vector<BindlessHandle> handles = ctx.GetBindlessHandles(bindings);
+
+    ctx.TransitionBindings(bindings);
+
+    std::filesystem::path shaderPath;
+    switch (GetParam())
+    {
+    case ShaderCompilerBackend::DXC:
+        shaderPath = VexRootPath / "tests/shaders/ScalarBlockLayoutTest.hlsl";
+        break;
+#if VEX_SLANG
+    case ShaderCompilerBackend::Slang:
+        shaderPath = VexRootPath / "tests/shaders/ScalarBlockLayoutTest.slang";
+        break;
+#endif
```

Comment:
I would actually force slang to be enabled for tests! So that we always test with it.
This is in the cmake tests.cmakelists.txt